src/frontend/src/lib/components/signer/SignerPermissions.svelte

```diff
@@ -0,0 +1,136 @@
+<script lang="ts">
+	import { IconWallet } from '@dfinity/gix-components';
+	import {
+		ICRC25_PERMISSION_GRANTED,
+		ICRC27_ACCOUNTS,
+		type IcrcScope,
+		type IcrcScopedMethod,
+		type PermissionsConfirmation
+	} from '@dfinity/oisy-wallet-signer';
+	import { isNullish, nonNullish } from '@dfinity/utils';
+	import { type ComponentType, getContext } from 'svelte';
+	import { fade } from 'svelte/transition';
+	import { icrcAccountIdentifierText } from '$icp/derived/ic.derived';
+	import IconAstronautHelmet from '$lib/components/icons/IconAstronautHelmet.svelte';
+	import IconShield from '$lib/components/icons/IconShield.svelte';
+	import SignerOrigin from '$lib/components/signer/SignerOrigin.svelte';
+	import ButtonGroup from '$lib/components/ui/ButtonGroup.svelte';
+	import { i18n } from '$lib/stores/i18n.store';
+	import { SIGNER_CONTEXT_KEY, type SignerContext } from '$lib/stores/signer.store';
+	import { toastsError } from '$lib/stores/toasts.store';
+	import { shortenWithMiddleEllipsis } from '$lib/utils/format.utils';
+	import { replaceOisyPlaceholders } from '$lib/utils/i18n.utils';
+
+	const {
+		permissionsPrompt: { payload, reset: resetPrompt }
+	} = getContext<SignerContext>(SIGNER_CONTEXT_KEY);
+
+	let scopes: IcrcScope[] | undefined;
+	$: scopes = $payload?.requestedScopes;
+
+	let confirm: PermissionsConfirmation | undefined;
+	$: confirm = $payload?.confirm;
+
+	/**
+	 * During the initial UX review, it was decided that permissions should not be permanently denied when "Rejected," but instead should be ignored.
+	 * This means that if the user selects "Reject," the permission will be requested again the next time a similar action is performed.
+	 * This approach is particularly useful since, for the time being, there is no way for the user to manage their permissions in the Oisy UI.
+	 */
+	const ignorePermissions = () => {
+		if (isNullish(confirm)) {
+			toastsError({
+				msg: { text: $i18n.signer.permissions.error.no_confirm_callback }
+			});
+			return;
+		}
+
+		confirm([]);
+
+		resetPrompt();
+	};
+
+	const approvePermissions = () => {
+		if (isNullish(confirm)) {
+			toastsError({
+				msg: { text: $i18n.signer.permissions.error.no_confirm_callback }
+			});
+			return;
+		}
+
+		confirm((scopes ?? []).map((scope) => ({ ...scope, state: ICRC25_PERMISSION_GRANTED })));
+
+		resetPrompt();
+	};
+
+	const onReject = () => ignorePermissions();
+
+	const onApprove = () => approvePermissions();
+
+	const listItems: Record<
+		IcrcScopedMethod,
+		{
+			icon: ComponentType;
+			label: string;
+		}
+	> = {
+		icrc27_accounts: {
+			icon: IconWallet,
+			label: replaceOisyPlaceholders($i18n.signer.permissions.text.icrc27_accounts)
+		},
+		icrc49_call_canister: {
+			icon: IconShield,
+			label: $i18n.signer.permissions.text.icrc49_call_canister
+		}
+	};
+
+	let requestAccountsPermissions = false;
+	$: requestAccountsPermissions = nonNullish(
```

Comment:
no need to add `scopes`, after `$:` here right?

`$: scopes, requestAccountsPermissions = nonNullish(....`

Reply:
No need indeed. Because `scopes` is used in the evaluation.